pkg/operator/ceph/cluster/osd/provision_spec.go

```diff
@@ -156,9 +158,15 @@ func (c *Cluster) provisionPodTemplateSpec(osdProps osdProperties, restart v1.Re
 	// we are passing false in case of osd and prepare pods because we don't want to overlap placement of osd on PVC's and non-PVC's.
 	p := cephv1.GetOSDPlacement(c.spec.Placement)
 	if osdProps.onPVC() {
+		// apply storageClassDeviceSet PreparePlacement
 		osdProps.getPreparePlacement().ApplyToPodSpec(&podSpec, false)
+		// apply placement.All() if it is not empty
+		if !reflect.DeepEqual(c.spec.Placement.All(), rookv1.Placement{}) {
```

Comment:
Isn't `ApplyToPodSpec()` already a no-op if the placement is empty? I think you could remove this check. Or did something not work without it?